### DISCUSSION TITLE
Why does keycloak use VARCHAR(36) instead of UUID for storage

### DISCUSSION THREAD

**Discussion by moaazassali** [keycloak/keycloak#26578] [Q&A]
After I connected keycloak to postgres database, I noticed that all the id columns are stored as VARCHAR(36). Given that UUID is supported natively in postgres and many other databases, why isn't that used instead? UUID is more efficient since it uses 16 bytes in both storage and RAM instead of 37 bytes in storage and 40 bytes in RAM for the VARCHAR(36) type.

**Comment by danielFesenmeyer** [comment]
I can't fully answer the question, but I'd like to mention that at least in older Keycloak versions for some API calls the realm name was set as realm ID - thus for some older realms the ID might NOT be a UUID.
